diff -urN TestFloat-2a/testfloat/arm-linux-gcc/Makefile TestFloat-2a-patched/testfloat/arm-linux-gcc/Makefile
--- TestFloat-2a/testfloat/arm-linux-gcc/Makefile	2015-03-10 00:55:24.647707865 -0400
+++ TestFloat-2a-patched/testfloat/arm-linux-gcc/Makefile	2015-03-10 00:54:57.483707406 -0400
@@ -7,7 +7,8 @@
 OBJ = .o
 EXE = .exe
 INCLUDES = -I. -I.. -I$(SOFTFLOAT_DIR)
-COMPILE_ASM = $(CROSS_COMPILE)gcc -c -o $@
+COMPILE_ASM = $(CROSS_COMPILE)gcc -c -o $@ \
+        -mhard-float -mfloat-abi=hard -mfpu=vfp
 COMPILE_C = $(CROSS_COMPILE)gcc -c -o $@ $(INCLUDES) -I- -O2 \
 	-mhard-float -mfloat-abi=hard -mfpu=vfp
 COMPILE_C_O0 = $(CROSS_COMPILE)gcc -c -o $@ $(INCLUDES) -I- -O1 \
